stores/migrations/sqlite/main/schema.sql

```diff
@@ -145,4 +145,42 @@ CREATE UNIQUE INDEX `idx_module_event_url` ON `webhooks`(`module`,`event`,`url`)
 
 -- dbObjectUserMetadata
 CREATE TABLE `object_user_metadata` (`id` integer PRIMARY KEY AUTOINCREMENT,`created_at` datetime,`db_object_id` integer DEFAULT NULL,`db_multipart_upload_id` integer DEFAULT NULL,`key` text NOT NULL,`value` text, CONSTRAINT `fk_object_user_metadata` FOREIGN KEY (`db_object_id`) REFERENCES `objects` (`id`) ON DELETE CASCADE, CONSTRAINT `fk_multipart_upload_user_metadata` FOREIGN KEY (`db_multipart_upload_id`) REFERENCES `multipart_uploads` (`id`) ON DELETE SET NULL);
-CREATE UNIQUE INDEX `idx_object_user_metadata_key` ON `object_user_metadata`(`db_object_id`,`db_multipart_upload_id`,`key`);
\ No newline at end of file
+CREATE UNIQUE INDEX `idx_object_user_metadata_key` ON `object_user_metadata`(`db_object_id`,`db_multipart_upload_id`,`key`);
+
+-- dbObject trigger to delete from slices
+CREATE TRIGGER delete_from_slices_after_objects_delete
+BEFORE DELETE ON objects
+BEGIN
+    DELETE FROM slices
+    WHERE slices.db_object_id = OLD.id;
+END;
+
+-- dbMultipartUpload trigger to delete from dbMultipartPart
+CREATE TRIGGER delete_from_multipart_parts_after_multipart_upload_delete
+BEFORE DELETE ON multipart_uploads
+BEGIN
+    DELETE FROM multipart_parts
+    WHERE multipart_parts.db_multipart_upload_id = OLD.id;
+END;
+
+-- dbMultipartPart trigger to delete from slices
+CREATE TRIGGER delete_from_slices_after_multipart_parts_delete
+BEFORE DELETE ON multipart_parts
+BEGIN
+    DELETE FROM slices
+    WHERE slices.db_multipart_part_id = OLD.id;
+END;
+
+-- dbSlices trigger to prune slabs
+CREATE TRIGGER delete_from_slabs_after_slice_delete
+AFTER DELETE ON slices
+BEGIN
+    DELETE FROM slabs
+    WHERE slabs.id = OLD.db_slab_id
+    AND slabs.db_buffered_slab_id IS NULL
```

Comment:
if this part of the condition can not be removed, why can we safely remove `AND slices.db_multipart_part_id IS NULL`

Reply:
If there was a slice apart from the one we just deleted that had `slices.db_multipart_part_id != NULL` we would find it in the `NOT EXIST` part of the query.

We only had to check for both the object id and the part id to be NULL before because we were LEFT JOINing but with the trigger we know exactly which slabs to look at.